src/patch.js

```diff
@@ -23,11 +25,67 @@ export default class Patch {
     return composedPatch.getChanges()
   }
 
+  static deserialize (serializedPatch) {
+    let {bytes, position} = serializedPatch
+    let buffer = new FlatBuffers.ByteBuffer(bytes)
+    buffer.setPosition(position)
+    let patch = Serialization.Patch.getRootAsPatch(buffer)
+    let changes = []
+    for (var i = 0; i < patch.changesLength(); i++) {
+      let serializedChange = patch.changes(i)
+      let oldStart = serializedChange.oldStart()
+      let newStart = serializedChange.newStart()
+      let oldExtent = serializedChange.oldExtent()
+      let newExtent = serializedChange.newExtent()
+      let change = {
+        oldStart: {row: oldStart.row(), column: oldStart.column()},
+        newStart: {row: newStart.row(), column: newStart.column()},
+        oldExtent: {row: oldExtent.row(), column: oldExtent.column()},
+        newExtent: {row: newExtent.row(), column: newExtent.column()}
+      }
+      let newText = serializedChange.newText()
+      let oldText = serializedChange.oldText()
+      if (newText != null) change.newText = newText
+      if (oldText != null) change.oldText = oldText
+
+      changes.push(change)
+    }
+
+    return new Patch({cachedChanges: changes})
```

Comment:
:+1:

In #8 I basically override `splice` with a function that throws an exception whenever a new `Patch` is instantiated with `cachedChanges`. I guess I can perform the same change here and then we can resolve the conflicts on the PR that gets merged last.